root/static/scripts/edit/URLCleanup.js

```diff
@@ -2941,11 +2941,11 @@ const CLEANUPS = {
     match: [new RegExp('^(https?://)?([^/]+\\.)?(spotify\\.com)/user', 'i')],
     type: LINK_TYPES.socialnetwork,
     clean: function (url) {
-      url = url.replace(/^(?:https?:\/\/)?(?:play|open)\.spotify\.com\/user\/([a-zA-Z0-9_-]+)\/?(?:[?#].*)?$/, 'https://open.spotify.com/user/$1');
+      url = url.replace(/^(?:https?:\/\/)?(?:play|open)\.spotify\.com\/user\/([^\/?#]+)\/?(?:[?#].*)?$/, 'https://open.spotify.com/user/$1');
       return url;
     },
     validate: function (url) {
-      return {result: /^https:\/\/open\.spotify\.com\/user\/[a-zA-Z0-9_-]+$/.test(url)};
+      return {result: /^https:\/\/open\.spotify\.com\/user\/[^\/?#]+$/.test(url)};
```

Comment:
There is a slightly different pattern for user’s identifier in [`lib/MusicBrainz/Server/Entity/URL/Spotify.pm`](https://github.com/metabrainz/musicbrainz-server/blob/v-2021-01-11/lib/MusicBrainz/Server/Entity/URL/Spotify.pm#L13). It should probably be streamlined.

Reply:
Would you suggest adding & to this one or removing it from the other?

Reply:
Just remove it please. `&` is separator between query parameters, thus it should normally be preceded with a `?` introducing the first query parameter, but some websites use `&` in place of `?`. That might explain the `&` in the pattern added in commit 4d92b8e1d7bfa30c91617f4f4b3525d31fd3f2b1. However there is no such URL for Spotify in the DB, so it is unlikely it exists in the wild.